src/main/java/com/unzer/payment/service/PaymentService.java

```diff
@@ -673,6 +682,25 @@ public Cancel cancelAuthorization(String paymentId, Cancel cancel)
         return cancel(cancel, urlUtil.getUrl(cancel));
     }
 
+    public Cancel cancelPreauthorization(String paymentId, Cancel cancel)
+            throws HttpCommunicationException {
+        cancel.setPaymentId(paymentId);
+        cancel.setTransactionType(Cancel.TransactionType.PREAUTHORIZE);
+        return cancel(cancel, urlUtil.getUrl(cancel));
+    }
+
+    public Cancel cancelPreauthorization(String paymentId, BigDecimal amount)
+            throws HttpCommunicationException {
+        Cancel cancel = new Cancel();
+        cancel.setAmount(amount);
+        return cancelAuthorization(paymentId, cancel);
```

Comment:
Shouldn't it be cancelPreauthorization here ?

Reply:
You are right, updated.